tests/conftest.py

```diff
@@ -15,10 +15,16 @@
 # For further info, check https://github.com/canonical/charmcraft
 
 import pathlib
+import tempfile
 
 import pytest
 
 
+@pytest.fixture(autouse=True, scope="session")
+def tmpdir_under_tmpdir(tmpdir_factory):
+    tempfile.tempdir = str(tmpdir_factory.getbasetemp())
```

Comment:
I was trying to understand how this works based on `tmpdir_factory` examples -- can all the tests using this fixture (which should be all of them due to `autouse=True`) access `tempfile.tempdir`?

Looking at docs, it seems as though when `tempfile.tempdir` is not `None`, it is only used as the `dir` default for other `tempfile` functions and it was unclear to me if it is useful outside of the context of the `tempfile` module: [1]

[1] https://docs.python.org/3/library/tempfile.html#tempfile.tempdir

Reply:
this change is exactly so charmcraft's calls to `tempfile.mkstemp` (in `logsetup`) use the test's tempdir instead of the default; this is the change that fixes #105.